File: Domain/Article.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain;

public class Article
{
    public int Id { get; set; }
    public string Title { get; set; }
    public Category Category { get; set; }
    public string Body { get; set; }
    public bool IsApproved { get; set; }
    [ForeignKey("Author")]
    public string AuthorId { get; set; }
    public AppUser Author { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    public ICollection<AppUser> Likes { get; set; } = new List<AppUser>();
    public ICollection<AppUser> Saves { get; set; } = new List<AppUser>();
}

File: client-app/src/app/models/article.ts
import { Category } from "./category";
import { Profile } from "./profile";

export interface Article {
  id: number;
  title?: string;
  body?: string;
  isApproved: boolean;
  image: string;
  author: Profile;
  likes: Profile[];
  category: Category;
  createdAt: string;
}

export interface ArticleFormValues {
  title: string;
  body: string;
  categoryId: number;
}
